maths/addition_without_arithmetic.nim

```diff
@@ -0,0 +1,36 @@
+## Illustrate how to add the integer without arithmetic operation
```

Comment:
```suggestion
## Illustrate how to implement addition of integers using bitwise operations
## See https://en.wikipedia.org/wiki/Bitwise_operation#Applications
```
I know this comes from the python version but the wording there to me is a bit weird (addition is an arithmetic operation, we are implementing here with bitwise operators instead of relying of the CPU instruction, which btw might be itself implemented with bitwise operators...). Also would add link to wikipedia which I think is relevant.

Reply:
Many thanks for the link. The wording seems much more relevant this way!